.github/workflows/pr.yaml

```diff
@@ -237,6 +237,36 @@ jobs:
         aws s3 cp bin/gitops s3://weave-gitops/gitops-${{matrix.os}}-${{steps.gitsha.outputs.sha}}
         aws s3 cp s3://weave-gitops/gitops-${{matrix.os}}-${{steps.gitsha.outputs.sha}} s3://weave-gitops/gitops-${{matrix.os}}
 
+  ci-publish-js-lib:
+    name: Publish js library
+    runs-on: ubuntu-latest
+    needs: [ci-js]
+    if: github.event_name == 'push'
+    permissions:
+      packages: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16.X"
+          registry-url: "https://npm.pkg.github.com"
+          scope: "@weaveworks"
+      - run: npm install
+      - run: make ui-lib
+      - name: Update package version
+        run: |
+          GITOPS_VERSION=$(git describe)
+          jq '.version = "'$GITOPS_VERSION'" | .name = "@weaveworks/weave-gitops-main"' < dist/package.json > dist/package-new.json
```

Comment:
Neato bandito, this will change the package name I think and we might have to alias in our build system (vs. changing all the imports from `import { foo } from @weaveworks/weave-gitops` -> `import { foo } from @weaveworks/weave-gitops-main`? Will do a quick test now..

Reply:
Oh, right. Didn't think about that :disappointed: 

If you want to test, there's a renamed package here: https://github.com/ozamosi/weave-gitops/packages/1590415

Reply:
This syntax seems to work

```
    "@weaveworks/weave-gitops-foo": "npm:@weaveworks/weave-gitops@v0.9.1-rc.1",
```

Will test it out with your repo. 🔌

Reply:
I heard you like `@`s..

```console
 yarn add @weaveworks/weave-gitops@npm:@ozamosi/weave-gitops-main@0.5.0-rc0.4.1-1708-g3568a017
```

Reply:
It seems to work! 🎉 , things are a bit out of sync right now so I'll just try and patch things up the interfaces and double check it all builds now..